modules/dnn/src/dnn.cpp

```diff
@@ -1611,7 +1620,32 @@ struct Net::Impl : public detail::NetImplBase
         else
             CV_Error(Error::StsNotImplemented, "Unknown backend identifier");
     }
+#ifdef HAVE_TENGINE
+
+    void initTengineBackend()
+    {
+        CV_TRACE_FUNCTION();
+
+        // Iterator to current layer.
+        MapIdToLayerData::iterator it = layers.begin();
+
+        for (; it != layers.end(); it++)
+        {
+            LayerData &ld = it->second;
+            Ptr<Layer> layerCur = ld.layerInstance;
 
+            if (!strcmp(ld.type.c_str(),"Convolution"))
```

Comment:
Why `Layer::finalize()` doesn't work here?
(or just the first run of `Layer::forward()`, check integration of `convolutionOp` from OpenCL backend)

Reply:
Move to Layer::forward . please checkout again